torch/_inductor/runtime/triton_heuristics.py

```diff
@@ -957,12 +978,38 @@ def __init__(
         self.compile_meta = compile_meta
         self.inductor_meta = inductor_meta
 
+    @staticmethod
+    def _serialize_metadata(metadata):
+        def is_namedtuple(obj) -> bool:
+            return (
+                isinstance(obj, tuple)
+                and hasattr(obj, "_asdict")
+                and hasattr(obj, "_fields")
+            )
+
+        if is_namedtuple(metadata):
+            return metadata._asdict()
+        else:
+            return metadata
+
+    @staticmethod
+    def _deserialize_metadata(metadata):
+        if isinstance(metadata, dict):
+            return TritonCompileResult._kernel_metadata_cls(tuple(metadata.keys()))(
+                **metadata
+            )
+        else:
+            return metadata
+
     def __getstate__(self) -> dict[str, Any]:
         kernel = self.kernel
         # replace the fields that don't pickle nicely
         kernel_state = {
             **kernel.__dict__,
-            "metadata": kernel.metadata._asdict(),
+            "metadata": self._serialize_metadata(kernel.metadata),
+            "packed_metadata": self._serialize_metadata(
```

Comment:
`packed_metadata` is annoying: for CPU Triton Backends, packed_metadata is the same unpicklable nested namedtuple type as metadata. For other backends, it's just a serializable tuple. Hence the reason to refactor into separate serialize/deserialize functions.

Reply:
Code reference: 
https://github.com/triton-lang/triton-cpu/blob/8b4787d60228422e7b5edce9f97a02c5df8ae079/python/triton/compiler/compiler.py#L368